src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -4753,22 +4753,32 @@ WorkerLimitCount(Node *limitCount, Node *limitOffset, OrderByLimitReference
 	if (workerLimitNode != NULL && limitOffset != NULL)
 	{
 		Const *workerLimitConst = (Const *) workerLimitNode;
-		Const *workerOffsetConst = (Const *) limitOffset;
-		int64 workerLimitCount = DatumGetInt64(workerLimitConst->constvalue);
-		int64 workerOffsetCount = DatumGetInt64(workerOffsetConst->constvalue);
-
-		workerLimitCount = workerLimitCount + workerOffsetCount;
-		workerLimitNode = (Node *) MakeIntegerConstInt64(workerLimitCount);
+		if (!workerLimitConst->constisnull)
+		{
+			/* Only update the worker limit if the const is not null.*/
```

Comment:
This comment appears to be about the next line, and is confusing. I think you want to move this before the `if` clause. Preferably preceded by an empty line.

Reply:
Sure, done. Both sytles seem to be used in this function, and both look ok to me.